### ISSUE TITLE
Keyboard trap while Page Explorer menu is open

### ISSUE THREAD

**Issue by kbayliss** [wagtail/wagtail#9325]
### Issue Summary

While the Page Explorer menu is open, the focus is trapped within the menu - you cannot continue tabbing to access links outside of the page menu.

### Steps to Reproduce

1. Open the Page Explorer
2. Keep pressing the 'tab' key, and you'll see the focus stays within the menu. This is inconsistent behaviour to all other menus in the sidebar.

### Technical details

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: Yes
-   Wagtail version: `4.0.2`
-   Browser version: Chrome Version 105.0.5195.125 (Official Build) (x86_64)

**Comment by thibaudcolas** [comment]
Hi @kbayliss, this is done on purpose, this menu is marked up as a `dialog`, which should trap focus. I believe all other menus should do the same in theory, and there is an open issue about it. Does that make sense to you?
